plugin/utils.py

```diff
@@ -65,6 +65,12 @@ def first(items: Iterable[T], test: Optional[Callable[[T], bool]] = None, defaul
     return next(filter(test, items), default)
 
 
+def fix_completion_syntax_highlight(view: sublime.View, point: int, code: str) -> str:
+    if view.match_selector(point, "source.php"):
+        return "<?php\n{}".format(code)
+    return code
```

Comment:
Do we see this being an issue with any other syntax definitions? Or is this unique only to PHP's need for <?php

Reply:
i don't know. this is one of my familiar language.

Reply:
The markdown case is more broken than I thought...
![image](https://user-images.githubusercontent.com/6594915/181285162-235d8f52-057a-4379-ba1b-68b9ef759513.png)

Reply:
<img width="510" alt="image" src="https://user-images.githubusercontent.com/32599364/181285406-7a4a25eb-36c2-441c-9d08-0415e7d20029.png">

Reply:
Probably this is something that is going to be extremely difficult to detect. We will need to inspect Scope and determine if it is in a code-fence

Reply:
> The markdown case is more broken than I thought... Copilot keeps adding trainling ```.
I think we should escape the complete text.  
For example, this is how I handle this for the phantom completion: https://github.com/TheSecEng/LSP-copilot/blob/e1fada07fc2ce4944b9dcfdfddfc2d5f7c2d8736/plugin/ui/completion.py#L325-L329

Notice `html.escape` part

Reply:
Is it necessary if we are already in fenced code block?

Reply:
I think it is necessary. Because it treats that ` ``` ` in the completion text as the closing part of the ` ```<lang>` block.

That's why you see ` ``` ` like outside of the highlighted region

Reply:
> html.escape

how is this revalent to `` ` ``?

and in my experiment, we don't have to escape text which is in fence code block.
![image](https://user-images.githubusercontent.com/6594915/181299249-9bf24730-b988-438d-b57b-29abf69a3f37.png)

Reply:
`html.escape` was just an example. In our case we should replace ` ` ` with a Unicode representation(if it even exists)

Reply:
but from my experiment, there is no need to use `html.escape` for a fenced code block.

I don't think this is what we want.
![image](https://user-images.githubusercontent.com/6594915/181303617-586bdf0c-5d86-4d6b-a1c0-d975438eafdf.png)

Reply:
ok great. a simple solution: just use ````` ```` ````` for fenced code block.

Reply:
yeah, we don't need `html.escape`, as `mdpopups` [already covers](https://github.com/facelessuser/sublime-markdown-popups/blob/master/st3/mdpopups/st_code_highlight.py#L88-L114) some basic HTML escaping. Sorry, there was a misunderstanding.

But I would give `code.replace("`", "&#96;")` a go, as it should fix [this](https://github.com/TheSecEng/LSP-copilot/pull/32#discussion_r931196040)

Reply:
> But I would give `` code.replace("`", "&#96;") `` a go, as it should fix [this](https://github.com/TheSecEng/LSP-copilot/pull/32#discussion_r931196040)

That would break `` ` `` in normal codes such as string literal in JS. A more elegant solution has been found above. :)